libtalek/client.go

```diff
@@ -39,6 +39,57 @@ type Client struct {
 	Rand    io.Reader
 }
 
+// msgHeader is a header prepended to each message fragment published by
+// libTalek to deal with fragmentation.
+// The fragment bit indicates that this is a subsequent message within
+// a sequence of messages. if a client is desynchronized, it can resume
+// a log once it finds the next message where the fragment bit is not set.
+type msgHeader struct {
+	flags byte // flags [0000000<Fragment>]
+	left  uint32
+}
+
+// Fill will fill a destination buffer with the header, followed by the src msg.
+func (m *msgHeader) Fill(dst, src []byte) int {
+	if len(dst) < 6 {
+		return -1
+	}
+	dst[0] = m.flags
+	binary.LittleEndian.PutUint32(dst[1:5], m.left)
+	num := uint32(copy(dst[5:], src[:]))
+	m.left -= num
+	m.flags |= 1
```

Comment:
If you're going to use a flags byte, it would be a good idea to use helper set/get methods, instead of doing manual byte manipulation every time. This line is hard to parse